backend/common/utils/spatial.py

```diff
@@ -0,0 +1,158 @@
+        bucket_name (str): The name of the S3 bucket.
+        asset_directory (str): The directory for the deep zoom assets.
+    """
+
+    def __init__(self, s3_provider: S3Provider = None):
+        self.s3_provider = s3_provider if s3_provider else S3Provider()
+        self.deployment_stage = os.getenv("DEPLOYMENT_STAGE", "staging")
+        # relying on dev bucket for rdev and test
+        self.env = "dev" if self.deployment_stage in ["rdev", "test"] else self.deployment_stage
+        self.bucket_name = f"spatial-deepzoom-{self.env}"
+        self.asset_directory = "spatial-deep-zoom"
+
+    def _calculate_aspect_ratio_crop(self, image_size):
+        """
+        Calculate the aspect ratio crop coordinates for an image.
+
+        Args:
+            image_size (int, int): The size of the image (width, height).
+
+        Returns:
+            tuple: The crop coordinates (left, upper, right, lower).
+        """
+        width, height = image_size
+        new_dimension = min(width, height)
+        left = (width - new_dimension) / 2
+        upper = (height - new_dimension) / 2
+        right = (width + new_dimension) / 2
+        lower = (height + new_dimension) / 2
+        return (left, upper, right, lower)
+
+    def _prepare_image(self, content):
+        """
+        Prepare the image array for processing.
+
+        Args:
+            content (dict): The content dictionary containing the image array.
+
+        Returns:
+            np.ndarray: The prepared image array.
+        """
+        resolution = "fullres" if "fullres" in content["images"] else "hires"
+        image_array = content["images"][resolution]
+        image_array_uint8 = np.uint8(image_array * 255 if resolution == "hires" else image_array)
+        return image_array_uint8
+
+    def _process_and_flip_image(self, image_array_uint8):
+        """
+        Process and flip the image.
+
+        Args:
+            image_array_uint8 (np.ndarray): The image array.
+
+        Returns:
+            np.ndarray: The processed and flipped image array.
+        """
+        Image.MAX_IMAGE_PIXELS = None  # Disable the image size limit
+        try:
+            with Image.fromarray(image_array_uint8) as img:
+                # Convert image to RGB mode if it's not already
+                if img.mode != "RGB":
+                    img = img.convert("RGB")
+                cropped_img = img.crop(self._calculate_aspect_ratio_crop(img.size))  # Crop the image
+                cropped_img.save(io.BytesIO(), format="JPEG", quality=90)  # Save or manipulate as needed
+                # Flip the image vertically due to explorer client rendering images upside down
+                flipped_img = cropped_img.transpose(Image.FLIP_TOP_BOTTOM)
+                return np.array(flipped_img)
+        except Exception as e:
+            logger.error(f"Error processing image: {e}")
+            raise Exception(f"Error processing image: {e}") from e
```

Comment:
Is there a reason we are raising a generic Exception from the previous exception?
```suggestion
            raise
```